packages/core/types/src/http/shipping-option/admin/payloads.ts

```diff
@@ -156,7 +156,14 @@ export interface AdminCreateShippingOption {
    * Learn more in the [Shipping Option](https://docs.medusajs.com/resources/commerce-modules/fulfillment/shipping-option#shipping-profile-and-types)
    * documentation.
    */
-  type: AdminCreateShippingOptionType
+  type?: AdminCreateShippingOptionType
+  /**
+   * The ID of the type of shipping option.
+   *
+   * Learn more in the [Shipping Option](https://docs.medusajs.com/resources/commerce-modules/fulfillment/shipping-option#shipping-profile-and-types)
+   * documentation.
+   */
+  type_id?: string
```

Comment:
suggestion: for the API, should we also do something similar to the DTO mutation? meaning type can be AdminCreateShippingOptionType | { id } or depending on previous comment AdminCreateShippingOptionType | string. In the first case we wouldnt even need any conversion i believe and mikro orm would handle it natively. I just find it a bit confusing that the shape and expectation changes all the time, also, what happen when both are provided? I believe only the id will be taken into account right? (but if we cant do that differently for now without creating too much breaking changes then lets go) 🙈

Reply:
I think we should be careful about supporting too many cases here. 

I would go with what @willbouch has done here at HTTP level, and apply it all the way down the stack.

That is:
```
type: DTO
type_id: string
```

If both are provided, I would throw:
```
_Only one of type and type_id can be provided_
```